packages/components/data-table/src/data-table.styles.tsx

```diff
@@ -107,6 +108,27 @@ const TableRow = styled.tr<TGetClickableRowStyleProps>`
     ${RowExpandCollapseButton} {
       opacity: 1;
     }
+
+    /* column divider showing up on hover  */
+    th:not(:first-of-type):after {
+      content: '';
+      position: absolute;
+      height: calc(
+        100% - 2 *
+          ${(props) =>
+            props.isCondensed
+              ? designTokens.marginForTableHeaderAsCondensed
+              : designTokens.marginForTableHeader}
+      );
```

Comment:
I'm using `calc` here for calculating the column divider height instead of using 100% height of the child element (and relying on the vertical margins) since in case one table header text is wrapped the height of the divider would be different than heights of all other dividers.